sklearn/decomposition/_nmf.py

```diff
@@ -1437,3 +1611,358 @@ def inverse_transform(self, W):
+
+        Returns
+        -------
+        W : ndarray of shape (n_samples, n_components)
+            Transformed data.
+
+        H : ndarray of shape (n_components, n_features)
+            Factorization matrix, sometimes called 'dictionary'.
+
+        n_iter_ : int
+            Actual number of iterations.
+
+        iter_offset : int, default=0
+            Number of previous iterations completed used for
+            initialization, only used in
+            :class:`sklearn.decomposition.MiniBatchNMF`.
+
+        A : array-like of shape (n_components, n_features)
+            Initial guess for the numerator auxiliary function
+
+        B : array-like of shape (n_components, n_features)
+            Initial guess for the denominator auxiliary function
+        """
+        check_non_negative(X, "NMF (input X)")
+        # check parameters
+        self._check_params(X)
+
+        if X.min() == 0 and self._beta_loss <= 0:
+            raise ValueError("When beta_loss <= 0 and X contains zeros, "
+                             "the solver may diverge. Please add small values "
+                             "to X, or use a positive beta_loss.")
+
+        n_samples, n_features = X.shape
+        # initialize or check W and H
+        W, H = self._check_w_h(X, W, H, update_H)
+
+        l1_reg_W, l1_reg_H, l2_reg_W, l2_reg_H = _compute_regularization(
+            self.alpha, self.l1_ratio, self.regularization)
+
+        # Initialize auxiliary matrices
+        A = H.copy()
+        B = np.ones(H.shape, dtype=H.dtype)
+
+        if self.solver == 'mu':
+            W, H, n_iter, iter_offset, A, B = _fit_multiplicative_update(
+                X, W, H, A, B, self._beta_loss, self._batch_size, 0,
+                self.max_iter, self.tol,
+                l1_reg_W, l1_reg_H, l2_reg_W, l2_reg_H,
+                update_H, self.verbose, self.forget_factor)
+        else:
+            raise ValueError("Invalid solver parameter '%s'." % self.solver)
+
+        if n_iter == self.max_iter and self.tol > 0:
+            warnings.warn("Maximum number of iterations %d reached. Increase "
+                          "it to improve convergence." % self.max_iter,
+                          ConvergenceWarning)
+
+        return W, H, n_iter, iter_offset, A, B
+
+    def partial_fit(self, X, y=None, **params):
+        is_first_call_to_partial_fit = not hasattr(self, 'components_')
+
+        if not is_first_call_to_partial_fit:
+            with config_context(assume_finite=True):
+                X = self._validate_data(X, accept_sparse=('csr', 'csc'),
+                                        dtype=[np.float64, np.float32],
+                                        reset=False)
+                # initialize W and H
+                H = self.components_
+                W = None
+                # Compute W given H and X using transform
+                W, *_ = self._fit_transform(X, H=H,
+                                            update_H=False)
```

Comment:
I think that with this change, MiniBatchNMF(warm_restarts=False).fit_transform(X) should be equivalent to MiniBatchNMF(warm_restarts=False).fit(X).transform(X). Still not true for warm_restarts=True but let's ignore that for now.

Reply:
Shouldn't be better to stick to the paper for now and open a new issue with warm start feature as a follow-up?

Reply:
The paper actually describes both options. I think it's important to enable warm restarts right away because it makes the minibatch iterations much faster (only 1 W update per iteration). Also I think it'll be the main usage for not too big datasets